### PR TITLE
feat: add ability to override Gateway container in devEnvironments

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
in DevEnvironments, there will be a new field `GatewayContainer`, which will have container definition, that will be used to override devworkpsace gateway container.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22036

### How to test this PR?

```bash
cat > /tmp/patch.yaml <<EOF
<PATCH_CONTENT>
EOF

chectl server:deploy \
     --installer operator \
     --platform <PLATFORM_TO_DEPLOY> \
     --che-operator-image <CUSTOM_OPERATOR_IMAGE> \
     --che-operator-cr-patch-yaml /tmp/patch.yaml
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Development resource are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-development-resources)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: controllers/devworkspace/solver/che_routing.go [hunk 1]
```diff
@@ -188,15 +189,21 @@ func (c *CheRoutingSolver) provisionPodAdditions(objs *solvers.RoutingObjects, c
...
+				corev1.ResourceMemory: resource.MustParse(constants.DefaultGatewayMemoryRequest),
+				corev1.ResourceCPU:    resource.MustParse(constants.DefaultGatewayCpuRequest),
 			},
 		},
-	})
+	}
+
+	if cheCluster.Spec.DevEnvironments.GatewayContainer != nil {
+		_ = deploy.CustomizeContainer(gatewayContainer, cheCluster.Spec.DevEnvironments.GatewayContainer)
+	}
```
Comment (amisevsk) [inline controllers/devworkspace/solver/che_routing.go:204]: Either `CustomizeContainer` should not return anything (currently it never returns anything but `nil`) or we shouldn't ignore the error here -- if a future update makes it sometimes return an error (e.g. by checking that request < limit) we probably won't notice a problem here until a bug is filed.


```suggestion
	if cheCluster.Spec.DevEnvironments.GatewayContainer != nil {
		if err := deploy.CustomizeContainer(gatewayContainer, cheCluster.Spec.DevEnvironments.GatewayContainer); err != nil {
			return err
		}
	}
```

**Review by amisevsk — APPROVED** [review]
LGTM, one comment.

**Comment by mkuznyetsov** [comment]
seems like happy-path test fails because of https://github.com/eclipse/che/issues/22163

**Pull request merged**